packages/components/src/local/react-table/index.tsx

```diff
@@ -135,17 +151,11 @@ export const ReactTable: React.FC<ReactTableProps> = (props) => {
    * @returns Row[]
    */
   const getFilteredRows = (appliedFilter: { col: string, filter: CellFilter[] }[]): Row[] => {
-    let filtered = rows
-    appliedFilter.forEach((filter: { col: string, filter: CellFilter[] }) => {
-      // collate list of acceptable values
-      const acceptableValues: string[] = filter.filter.map(f => f.label)
-      // update set of filtered rows
-      filtered = filtered.filter((row: Row) => {
-        // see if this value is in the acceptable values
-        return acceptableValues.includes(row[filter.col])
-      })
+    const acceptableValues: string[] = flattenDeep(appliedFilter.map(filter => filter.filter.map(f => f.label)))
```

Comment:
Ooh @lebaphi, this is good & clever "js-array" code.  But, I'm worried about the logic.  I may have this wrong, but if the user selects "Closed" in the "Status" column, it will check for "Closed" in other columns.  Aaah, is it doing a compare for every field in every row with the flattened list?

That's quite inefficient if the player has, for example, just asked to see "Status":"Pending" messages.

Happy to discuss :-)

Reply:
Yeah, I see what you mean, I will update a bit the logic 👍